maven.grammar/src/org/netbeans/modules/maven/hyperlinks/HyperlinkProviderImpl.java

```diff
@@ -453,5 +215,331 @@ private FileObject getPath(FileObject parent, String path) {
+                    int ac = after.indexOf("}");//NOI18N
+                    if (bo > bc && ac > -1 && (ac < ao || ao == -1)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+        
+        boolean isMavenDependency() {
+            return artifactId != null && groupId != null && version != null;
+        }
+        
+        boolean isHyperlinkPoint() {
+            return (isHyperlinkUrl() || isFileSystemLink() || isMavenProperty() || isMavenDependency());
+        }
+        
+        private void calculateInfo(Token<XMLTokenId> token, TokenSequence<XMLTokenId> xml) {
+            isText = token.id() == XMLTokenId.TEXT;
+            if (isText) {
+                ftokenOff = xml.offset();
+                ftext = token.text().toString();
+                
+                if (projectFileObject != null && getPath(projectFileObject, ftext) != null) {
+                    xml.movePrevious();
+                    token = xml.token();
+                    if (token != null && token.id().equals(XMLTokenId.TAG) && TokenUtilities.equals(token.text(), ">")) {//NOI18N
+                        xml.movePrevious();
+                        token = xml.token();
+                        if (token != null && token.id().equals(XMLTokenId.TAG)) {
+                            if (TokenUtilities.equals(token.text(), "<module")) {//NOI18N
+                                if (!ftext.endsWith("/pom.xml")) {
+                                    ftext += "/pom.xml"; //NOI18N
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    xml.movePrevious();
+                    token = xml.token();
+                    if (token != null && token.id().equals(XMLTokenId.TAG) && TokenUtilities.equals(token.text(), ">")) { //NOI18N
+                        xml.movePrevious();
+                        token = xml.token();
+                        if (TokenUtilities.equals(token.text(), "<artifactId") || //NOI18N
+                            TokenUtilities.equals(token.text(), "<groupId") || //NOI18N
+                            TokenUtilities.equals(token.text(), "<type") || //NOI18N
+                            (TokenUtilities.equals(token.text(), "<version") && !ftext.startsWith("${"))) { //NOI18N
+                            resetSequenceToDependencyTagToken(xml);
+                            if (TokenUtilities.equals(xml.token().text(), "<dependency")) {
+                                while(!TokenUtilities.equals(xml.token().text(), "</dependency")) { //NOI18N
+                                    xml.moveNext();
+                                    token = xml.token();
+                                    if (TokenUtilities.equals(token.text(), "<artifactId")) { //NOI18N
+                                        xml.moveNext();
```

Comment:
What if the token +2 is an attribute / value ? It is an invalid POM, but could happen. There could also be  an excess whitespace before '>' token.
Maybe XMLSyntaxSupport from xml.text could help to navigate better